src/Sylius/Behat/Context/Api/Admin/ManagingShippingCategoriesContext.php

```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * @When I browse shipping categories
+     */
+    public function iBrowseShippingCategories(): void
+    {
+        $this->client->index('shipping_categories');
+    }
+
+    /**
+     * @When I check (also) the :shippingCategoryName shipping category
+     */
+    public function iCheckTheShippingCategory(string $shippingCategoryName): void
+    {
+        Assert::true($this->client->hasItemWithValue('name', $shippingCategoryName));
+    }
+
+    /**
+     * @When I do not specify its code
+     * @When I specify its code as :shippingCategoryCode
+     */
+    public function iSpecifyItsCodeAs(string $shippingCategoryCode = ''): void
+    {
+        if ($shippingCategoryCode !== '') {
+            $this->client->addRequestData('code', $shippingCategoryCode);
+        }
+    }
+
+    /**
+     * @When I name it :shippingCategoryName
+     * @When I do not specify its name
+     */
+    public function iNameIt(string $shippingCategoryName = ''): void
+    {
+        if ($shippingCategoryName !== '') {
+            $this->client->addRequestData('name', $shippingCategoryName);
+        }
+    }
+
+    /**
+     * @When I specify its description as :shippingCategoryDescription
+     */
+    public function iSpecifyItsDescriptionAs(string $shippingCategoryDescription): void
+    {
+        $this->client->addRequestData('description', $shippingCategoryDescription);
+    }
+
+    /**
+     * @Then I should be notified that shipping category with this code already exists
+     */
+    public function iShouldBeNotifiedThatShippingCategoryWithThisCodeAlreadyExists(): void
+    {
+        Assert::same($this->client->getError(), 'code: The shipping category with given code already exists.');
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatCodeIsRequired(string $element): void
+    {
+        Assert::same(
+            $this->client->getError(), sprintf('%s: Please enter shipping category %s.', $element, $element)
+        );
+    }
+
+    /**
+     * @Then I should see a single shipping category in the list
+     * @Then I should see :numberOfShippingCategories shipping categories in the list
+     */
+    public function iShouldSeeShippingCategoriesInTheList(int $numberOfShippingCategories = 1): void
+    {
+        $this->client->index('shipping_categories');
+        Assert::same($this->client->countCollectionItems(), $numberOfShippingCategories);
+    }
+
+    /**
+     * @Then /^the (shipping category "([^"]+)") should be in the registry$/
+     * @Then /^the (shipping category "([^"]+)") should appear in the registry$/
+     */
+    public function theShippingCategoryShouldAppearInTheRegistry(ShippingCategoryInterface $shippingCategory): void
+    {
+        $this->client->index('shipping_categories');
+        Assert::true($this->client->hasItemWithValue('name', $shippingCategory->getName()));
```

Comment:
Please, add custom exception message, for the `true`/`false` assertions especially.